lib/libc/Kconfig

```diff
@@ -36,7 +36,7 @@ config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
 	depends on ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || (NATIVE_LIBRARY)
-	depends on !(CPP && ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y"))
+	depends on !REQUIRES_FULL_LIBCPP || ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y")
```

Comment:
@keith-packard - I'm not sure with your last comment `it will work as-is`  if you want this changed.  Please let me know, otherwise I can mark this conversation resolved.